Ui tests

Fatalll opened the pull request.

Fatalll committed:
UI tests initial commit

Fatalll committed:
Merge remote-tracking branch 'remotes/origin/master' into ui-tests

Fatalll committed:
Merge branch 'ui-fixes' into ui-tests

Fatalll committed:
Merge remote-tracking branch 'remotes/origin/master' into ui-tests

TanVD commented on src/test/kotlin/tanvd/grazi/ide/ui/GraziGuiTest.kt:
```diff
@@ -0,0 +1,27 @@
+import org.junit.Test
+
+
+@RunWithIde(CommunityIde::class)
```
By the way, we are using IU....

Fatalll replied:
Yeah

Fatalll committed:
First real tests + new menu

TanVD committed:
Move part of Gui tests logic to buildSrc.
Simplify it a bit.
Also, execute `clean` before each `compileKotlin` (needed to clean
sandbox before test or IDE start)

TanVD committed:
Add a directory with reports as an artifact

TanVD committed:
Fix indentation in CircleCI config

TanVD committed:
Add reports also for release. Remove unneeded stacktrace param

Fatalll committed:
UI tests done

TanVD committed:
Separate plugin and gui-tests for it

TanVD committed:
Fixes for ui-tests

TanVD committed:
Fixes for UI tests, reworkes circle-ci

TanVD committed:
Fixes to CircleCI config and uncomment UI tests

Fatalll committed:
Even better tests
